### PR TITLE
MediaLoader on Live radio page

### PR DESCRIPTION
**Pull request by amoore108**
Resolves JIRA https://jira.dev.bbc.co.uk/browse/WSTEAM1-1288

Overall changes
======
- Removes iframe media player from Live Radio page in favour of new MediaLoader
- Adds new `liveRadio` config to MediaLoader to create SMP config for Live Radio streams
- Uplifts LiveRadioPage to TS

Testing
======
1. Visit http://localhost.bbc.com:7080/hausa/bbc_hausa_radio/liveradio?renderer_env=live
2. Confirm media player renders and allows playback, showing all related controls for Live content

Helpful Links
======
_Add Links to useful resources related to this PR if applicable._

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

### CODE REVIEW

File: src/app/routes/liveRadio/getInitialData/index.js [hunk 1]
```diff
@@ -44,6 +44,13 @@ export default async ({ path: pathname, pageType, service, toggles }) => {
 
     const getRadioScheduleData = path(['radioScheduleData']);
 
+    const mediaLoaderBlock = [
```
Comment (karinathomasbbc) [inline src/app/routes/liveRadio/getInitialData/index.js]: For consistency with our other page types (work currently in progress: #11928), can we rename this to `mediaBlock`?

**Review by karinathomasbbc — APPROVED** [review]

**Comment by karinathomasbbc** [comment]
I am surprised that the cypress tests for live radio are not failing... and having a glance, they are not really testing the audio player properly: https://github.com/bbc/simorgh/blob/4f1dffd50d87707f52c92ae4fa8e7e1192c94029/cypress/e2e/pages/liveRadio/tests.js#L24-L50, but rather checking that the embed URL in the data is reachable.

I am wondering if we should wait for the work in #11928 to be completed, because they are refactoring the media player code from the live page cypress tests, which would make it available to use here.

**Comment by amoore108** [comment]
> I am surprised that the cypress tests for live radio are not failing... and having a glance, they are not really testing the audio player properly:
> 
> https://github.com/bbc/simorgh/blob/4f1dffd50d87707f52c92ae4fa8e7e1192c94029/cypress/e2e/pages/liveRadio/tests.js#L24-L50
> 
> , but rather checking that the embed URL in the data is reachable.
> I am wondering if we should wait for the work in #11928 to be completed, because they are refactoring the media player code from the live page cypress tests, which would make it available to use here.

Yea somewhat fortunate in a way! We could wait if the code is portable between the page types, although I guess we're in no worse a position now than we were before? Wouldn't say that at any other time than now, but given the timeframe 👀

**Comment by karinathomasbbc** [comment]
> > , but rather checking that the embed URL in the data is reachable.
> > I am wondering if we should wait for the work in #11928 to be completed, because they are refactoring the media player code from the live page cypress tests, which would make it available to use here.
> 
> Yea somewhat fortunate in a way! We could wait if the code is portable between the page types, although I guess we're in no worse a position now than we were before? Wouldn't say that at any other time than now, but given the timeframe 👀

It looks like the live page media tests have not been refactored, just used as inspiration :) https://github.com/bbc/simorgh/blob/52914a19ab91c68e9b751157e2258fbc4e78de3f/cypress/e2e/pages/onDemandTV/tests.js#L29

So, could we update the cypress tests for the live radio page to check for the existence of the media loader, instead of checking the embed URL?

**Comment by amoore108** [comment]
> So, could we update the cypress tests for the live radio page to check for the existence of the media loader, instead of checking the embed URL?

Good shout, I've updated the e2e test now to check the existence of the medialoader container.

File: cypress/e2e/pages/liveRadio/tests.js [hunk 2]
```diff
@@ -27,23 +25,9 @@ export default ({ service, pageType, variant }) =>
...
-        });
-
-        it('embed URL should be reachable', () => {
-          cy.testResponseCodeAndTypeRetry({
-            path: embedUrl,
-            responseCode: 200,
-            type: 'text/html',
-            allowFallback: true,
+        it('should render a valid media player', () => {
+          cy.request(getDataUrl(Cypress.env('currentPath'))).then(() => {
```
Comment (karinathomasbbc) [inline cypress/e2e/pages/liveRadio/tests.js]: Looks like we can remove this line
```suggestion
```

File: src/app/components/MediaLoader/index.tsx [hunk 3]
```diff
@@ -173,9 +182,12 @@ const MediaLoader = ({ blocks, embedded, className }: Props) => {
     showAdsBasedOnLocation,
   } = useContext(RequestContext);
 
-  const showPlaceholder = !PAGETYPES_IGNORE_PLACEHOLDER.includes(pageType);
+  const PAGETYPE_SUPPORTS_PLACEHOLDER =
+    !PAGETYPES_IGNORE_PLACEHOLDER.includes(pageType);
```
Comment (LilyL0u) [inline src/app/components/MediaLoader/index.tsx:186]: Does 'IGNORE_PLACEHOLDER' mean it does not have a placeholder image?

File: src/app/components/MediaLoader/utils/buildSetting.test.ts
Comment (LilyL0u) [inline src/app/components/MediaLoader/utils/buildSetting.test.ts:1]: Rename to buildSettings (with the s on the end) to keep consistent with buildSettings.ts?

Comment (amoore108) [reply src/app/components/MediaLoader/utils/buildSetting.test.ts:1]: Oh good spot! Shall rename this

**Review by LilyL0u — APPROVED** [review]

**Review by HarveyPeachey — APPROVED** [review]

**Pull request merged**